_usage/interoperability.md

```diff
@@ -0,0 +1,109 @@
+ 
+Terminology note: in this document, *TDS* and *T-SQL*, and *PostgreSQL* and *PostgreSQL SQL* are used as synonyms for the type of SQL dialect as well as the corresponding connection to the database.
+
+In the context of this document, the Babelfish database is a PostgreSQL database created for the purpose of emulating the SQL Server multi-DB environment in the T-SQL dialect, e.g. with a `master` and `tempdb` database and one or more user databases. In Aurora PostgreSQL, this database is always named `babelfish_db`. A TDS connection will always be placed inside this Babelfish database, even though that is not visible through T-SQL. There can only be a single Babelfish database per PostgreSQL instance.
+
+Since the Babelfish database is ultimately a PostgreSQL database, it is technically possible to access it from the PostgreSQL port and run PostgreSQL SQL statements against it. However, users should tread carefully when accessing the Babelfish database through PostgreSQL, as there are various complexities to be considered. 
+
+From a Babelfish perspective, it is simplest to avoid accessing the Babelfish database from PostgreSQL altogether. However, there may be valid reasons for wanting to use PostgreSQL functionality anyway. For example, when a particular aspect of T-SQL is not supported by Babelfish, it might be possible to implement that aspect natively in PostgreSQL so as to allow migrating the application to Babelfish PostgreSQL. However, this comes with certain implications and risks, as described below.
+
+The interoperability considerations apply whenever the execution context is not pure TDS/T-SQL, i.e. when PostgreSQL SQL statements are executed in a Babelfish context. This is the case when:
+
+- In a PostgreSQL connection, PostgreSQL statements are executed against Babelfish-created objects (e.g. tables, views, procedures). 
+- In a TDS connection, T-SQL statements are executed against PostgreSQL-created objects (e.g. tables, views, procedures) or when PostgreSQL built-in functions are called from T-SQL. Based on customer feedback, this seems to be the more common of these scenarios.
+
+In both cases, this effectively means the SQL dialect is temporarily (and transparently) switched in the session while accessing/executing the objects created in a different SQL dialect (this does not apply to accessing a table).
+
+When one of the above cases occurs in a Babelfish-based application, one or more of the following interoperability aspects may need to be considered. It is the responsibility of the Babelfish user to determine whether an application with such a mixed-SQL dialect scenario works as expected. Also, the user should anticipate potential issues when upgrading to future versions of Babelfish; such issues may need to be resolved by the user themselves, as AWS may not be able to support such mixed-SQL dialect cases.
```

Comment:
"AWS may not be able...' ==> should we mention AWS in the OSS docs? not sure how to reword this

Reply:
How about truncating the sentence (just taking off the 'why' since we're open-source and not counting on AWS support):

Also, the user should anticipate potential issues when upgrading to future versions of Babelfish; such issues may need to be resolved by the user themselves.

Reply:
There was a reference to 'Aurora' as well - I've updated the sentence to remove the reference.